### ISSUE TITLE
Erratic Dilation behavior

### ISSUE THREAD

**Issue by staticfloat** [FluxML/NNlib.jl#44]
Uh-oh, I should have tested this more thoroughly looks like.  :/

```julia
# 2-length convolution with 1 channel in and 1 channel out
w = reshape(Float64[1, 1], (2, 1, 1))

# 128-length input vector with 1 channel and 1 batch.
x = reshape(Float64[1:128...], (128, 1, 1))

# Let's calculate the same thing a couple different times
ys = []
for idx in 1:10
    push!(ys, NNlib.conv(x, w; dilation = 2))
end

# Do some of these randomly have a bunch of NaN's in them?  Why yes they do:
[any(isnan.(ys[idx])) for idx in 1:10]
```

Running the above results in:
```
10-element Array{Bool,1}:
 false
  true
  true
 false
 false
 false
 false
 false
  true
  true
```

The fact that this is erratic makes me think the BLAS routine is reading from uninitialized/clobbered memory.

**Comment by staticfloat** [comment]
@tejank10 sorry to so quickly have to come back to you about this.

**Comment by tejank10** [comment]
@staticfloat, no problem. I changed the code on line 178 of `impl/conv.jl` to `size(w)` instead of `dilation_dims(w)` and it didn't give any error for the above inputs (I ran the loop for about 10k times). But what worries me is that how could it (the code with `dilation_dims`) give different output for the same input, and can also gives the correct output in some cases?

**Comment by staticfloat** [comment]
Fixed by #94
